util/blocktime.py

```diff
@@ -133,6 +133,68 @@ def timeSinceTimestamp(self, block_i):
     return int(block_i)
 
 
+@enforce_types
+def ethTimestamptoBlock(chain, timestamp: Union[float, int]) -> int:
+    """Example: 1648872899.0 --> 4928"""
+    current_block = chain.height
+    current_time = chain.time()
+
+    return ethCalcBlockNumber(current_time, current_block, timestamp, chain)
+
+
+@enforce_types
+def ethCalcBlockNumber(ts, block, target_ts, chain):
+    AVG_BLOCK_TIME = 12  # seconds
+    diff = target_ts - ts
+    diff_blocks = diff // AVG_BLOCK_TIME
+    block += diff_blocks
+
+    ts_found = chain[block].timestamp
+    if abs(ts_found - target_ts) > 12 * 5:
+        return ethCalcBlockNumber(ts_found, block, target_ts, chain)
+
+    return block
+
+
+@enforce_types
+def ethFindFirstThuBlock(chain, block_number: int) -> int:
+    """
+    @arguments
+        chain -- brownie.networks.chain
+        block_number -- int
+    @return
+        block_number -- int
+    @description
+        Finds the first Thursday block closest to the given block number
+    """
+
+    def get_tsdtday(block_i):
+        block_ts = chain[block_i].timestamp
+        block_dt = datetime.fromtimestamp(block_ts)
+        block_day = block_dt.weekday()
+        return block_ts, block_dt, block_day
```

Comment:
This function returns three items, though all of its callers only use the third item.

Also, the function name is cryptic (a concatenation of all return items?). Probably to fix: only return the third item, then rename accordingly (and less cryptic).